homeassistant/components/unifi/switch.py

```diff
@@ -158,6 +160,16 @@ async def async_port_forward_control_fn(
     await hub.api.request(PortForwardEnableRequest.create(port_forward, target))
 
 
+async def async_traffic_rule_control_fn(
+    hub: UnifiHub, obj_id: str, target: bool
+) -> None:
+    """Control traffic rule state."""
+    traffic_rule = hub.api.traffic_rules[obj_id].raw
+    await hub.api.request(TrafficRuleEnableRequest.create(traffic_rule, target))
+    # Update the traffic rules so the UI is updated appropriately
+    await hub.api.traffic_rules.update()
```

Comment:
I guess since using the data update coordinator it could be used here to force an update and reset the timer to keep it aligned from "last" call, I guess this would warrant one update coordinator per api then